server/server.ts

```diff
@@ -109,8 +108,20 @@ if (process.env.NODE_ENV === 'production') {
 app.use(deduplicateSlash());
 app.use(noSlash());
 app.use(compression());
-app.use(bodyParser.json({ limit: '50mb' }));
-app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }));
+app.use(express.json({ limit: '50mb' }));
+app.use(express.urlencoded({ limit: '50mb', extended: true }));
+// needed to parse analytics events from the client sent with `navigator.sendBeacon`
+app.use(express.text(), (req, res, next) => {
```

Comment:
Should this be route-level middleware instead? I'm not sure any other cases where we send text to the server but could there be a performance penalty here?

Reply:
good point!